gobblin-data-management/src/main/java/gobblin/data/management/copy/CopyableDatasetBase.java

```diff
@@ -0,0 +1,11 @@
+package gobblin.data.management.copy;
+
+import gobblin.dataset.Dataset;
+
+
+/**
+ * A common superinterface for {@link Dataset}s that can be operated on by distcp.
+ * Concrete classes must implement a subinterface of this interface ({@link CopyableDataset} or {@link IterableCopyableDataset}).
+ */
+interface CopyableDatasetBase extends Dataset {
```

Comment:
Type awareness. `CopySource` gets datasets of type `CopyableDataset` or `IterableCopyableDataset`, it doesn't care about any other type of dataset.